packages/components/.storybook/tokens.stories.mdx

```diff
@@ -12,28 +12,41 @@ import styles from "./tokens.module.css";
 ## EDS Colors
 
 export const mapEDSColorToTokenArray = (color) => {
-  return Object.keys(tokens.eds.color[color])
+  // Handle nested color names
+  const colorKeys = color.split("-");
+  const colorTokens = colorKeys.reduce((obj, key) => {
+    return obj[key];
+  }, tokens.eds.color);
+  return Object.keys(colorTokens)
```

Comment:
Gonna be honest – it took me a while to figure out what you were doing in this reduce. I feel like there should be a pattern that's a little easier to reason about for this situation. At this point, it looks to me like you could just use the last value in the `colorKeys` array like
```
const colorToUse = colorKeys[colorKeys.length - 1];
return Object.keys(tokens.eds.color[colorToUse])
```
or something?

Reply:
Could also extract some functions or utils (not sure how that works with mdx)

```js
const colorTokens = getTokensForColor(color);
return Object.keys(colorTokens);
```

Reply:
the tokens file (`variables-nested.json`) looks like:
```
"brand": {
    ...
    "700": "#433BDE"
},
"grading": {
    "yellow": {
        "100": "#FDF1D0",
        "500": "#FFC55B"
    }
 },
```

so the key `a-b` means we need to access `tokens[a][b]`. It's easier if I assume there's max 1 level of nesting that's going to happen, but thought it _might_ be useful to generalize.

Can def extract functions though (and maybe add comments? haven't tried that)

Reply:
could be a hasty abstraction though, to handle just 1 extra level of nesting it could be
```
let colorTokens = tokens.eds.color[colorKeys[0]];
if colorKeys.length === 2 {
    colorTokens = colorTokens[colorKeys[1]]
}
```